server/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -263,18 +266,23 @@ private static Analyzer buildCustomAnalyzer(AnalyzeAction.Request request, Analy
             }
         }
 
-        CustomAnalyzer customAnalyzer = null;
-        if (analyzer instanceof CustomAnalyzer) {
-            customAnalyzer = (CustomAnalyzer) analyzer;
-        } else if (analyzer instanceof NamedAnalyzer && ((NamedAnalyzer) analyzer).analyzer() instanceof CustomAnalyzer) {
-            customAnalyzer = (CustomAnalyzer) ((NamedAnalyzer) analyzer).analyzer();
+        Analyzer customAnalyzer = null;
```

Comment:
Can you use AnalysisComponentsProvider here instead? Avoids casting lower down, and makes the instanceof checks simpler.

Reply:
+1, I pushed something, needed to change some internal method signatures for it to work but it looks better to me as well. Lets wait for CI to see if I accidentally broke sth., this code is a bit fragile.